static/js/click_handlers.js

```diff
@@ -504,6 +504,29 @@ exports.initialize = function () {
         compose_actions.respond_to_message({trigger: 'reply button'});
     });
 
+    $('.button.dropdown').click(function () {
```

Comment:
The jquery state toggling logic here should probably a minimuum be refactored into another module rather than living in the main click event handler registration. Using the presence of a class on a div for tracking UI state is something I know people have different opinions about, with jQuery (vs react or vue) I sort of like to keep the UI state in CSS rather than tracking separately, so the DOM is the authoritative UI state source.

Reply:
Yeah, I would probably put this in `static/js/components.js` as `components.dropdown()`; and then this line could be e.g.

`components.dropdown($(".button.dropdown"))`

(Or maybe using a better selector; I think we probably actually want the `$(body).on("click", ".button.dropdown", ...` pattern so that we can use this in HTML segments that are rendered dynamically).

Reply:
Yup, all makes good sense, that said, I think as per @rishig we probably want to re-work this to use the popover code, so most of this will go away anyway, blocking on https://github.com/zulip/zulip/pull/11804 merging before proceeding with that